smartsheet-csharp-sdk/main/Smartsheet/Api/Internal/Http/DefaultHttpClient.cs

```diff
@@ -223,28 +257,70 @@ public virtual HttpResponse Request(HttpRequest smartsheetRequest)
-
+                //httpClient.BaseUrl = new Uri(smartsheetRequest.Uri.GetLeftPart(UriPartial.Authority));
+                Uri baseUrlAsURI = new Uri(smartsheetRequest.Uri.GetLeftPart(UriPartial.Authority));
+                RestClientOptions options = new RestClientOptions(baseUrlAsURI);
+                options.FollowRedirects = true;
+                this.httpClient = new RestClient(options, null, null, true);
                 Stopwatch timer = new Stopwatch();
 
                 // Make the HTTP request
                 timer.Start();
-                restResponse = httpClient.Execute(restRequest);
+                Task<RestResponse> restResponseAsTask = this.httpClient.ExecuteAsync(restRequest);
+                restResponseAsTask.Wait();
+                restResponse = restResponseAsTask.Result;
                 timer.Stop();
 
                 LogRequest(restRequest, restResponse, timer.ElapsedMilliseconds);
 
                 if (restResponse.ResponseStatus == ResponseStatus.Error)
                 {
-                    throw new HttpClientException("There was an issue connecting.");
+                    Boolean restClientDebugging = false;
+                    if (restClientDebugging) {
+                        StringBuilder builder = new StringBuilder();
+                        builder.Append(" Smartsheet request URI is " + smartsheetRequest.Uri + "\n");
+                        foreach (KeyValuePair<string, string> header in smartsheetRequest.Headers)
+                        {
+                            builder.Append("Headers: " + "\n");
+                            builder.Append("Key: " +  header.Key + " Value: " + header.Value + "\n");
+                        }
+                        builder.Append(" baseUrlAsURI is " + baseUrlAsURI.ToString() + "\n");
+                        builder.Append(" Smartsheet request method is " + smartsheetRequest.Method + "\n");
+                        builder.Append(" Was there a body added??? " + bodyAdded.ToString() + "\n");
+                        if (smartsheetRequest.Entity != null) {
+                            builder.Append(" Was there a body entity content type: " + smartsheetRequest.Entity.ContentType + "\n");
+                        builder.Append(" Was there a body entity as string : " + smartsheetRequest.Entity.GetContentAsString() + "\n");
+                        }
+
+                        builder.Append(" There was an issue connecting.");
+                        builder.Append(" RestResponse code is " + restResponse.StatusCode.ToString() + "\n");
+                        builder.Append(" RestResponse ErrorMessage is " + restResponse.ErrorMessage + "\n");
+                        builder.Append(" RestResponse Content is " + restResponse.Content + "\n");
+
+                    throw new HttpClientException(builder.ToString());
+                    } else {
+                        throw new HttpClientException("There was an issue connecting.");
```

Comment:
should this include the error message if there is one?

Reply:
Have added one now 👍